weave/tests/test_uploader.py

```diff
@@ -1347,3 +1348,48 @@ def test_upload_correct_version_number(test_basket):
         manifest_dict = json.load(file)
 
     assert manifest_dict["weave_version"] == weave.__version__
+
+def test_upload_from_s3fs(test_basket):
+    """Test that a basket can be uploaded from s3fs to the local file
+    system or s3fs.
+    """
+    pantry_path = os.path.join(test_basket.pantry_path, "test-pantry-1")
+
+    test_basket.file_system.mkdir(pantry_path)
+
+    tmp_txt_file = test_basket.tmpdir.join("test.txt")
+    tmp_txt_file.write("this is a test")
+
+    nested_path = os.path.join(pantry_path, "text.txt")
+    file_to_move = os.path.join(test_basket.pantry_path, "text.txt")
+
+    if test_basket.file_system == local_fs:
+        minio_path = "test-source-file-system"
+        s3fs.mkdir(minio_path)
+        file_to_move = os.path.join(minio_path, "test.txt")
+
+    # Must upload a file because Minio will remove empty directories
+    if test_basket.file_system == local_fs:
+        s3fs.upload(str(tmp_txt_file.realpath()), file_to_move)
+    else:
+        test_basket.file_system.upload(str(tmp_txt_file.realpath()), file_to_move)
+    test_basket.file_system.upload(str(tmp_txt_file.realpath()), nested_path)
+    # Make the Pantries.
+    pantry_1 = Pantry(
+        IndexPandas,
+        pantry_path=pantry_path,
+        file_system=test_basket.file_system,
+    )
+    basket_uuid = pantry_1.upload_basket(
+                           upload_items=[{'path': file_to_move,'stub':False}],
+                           source_file_system=s3fs,
+                           basket_type="test_s3fs_upload"
+                           )["uuid"][0]
+
+    basket = pantry_1.get_basket(basket_uuid)
+
+    if not test_basket.file_system == local_fs:
+        assert basket.ls()[0].endswith('text.txt')
+    else:
+        assert basket.ls()[0].endswith('test.txt')
+        s3fs.rm(minio_path,recursive=True)
```

Comment:
I think you only need the s3fs.rm() inside the if statement, the asserts should be outside

Reply:
wait, why does it have a different name? shouldn't it be the same? is there a way to fix this?